utoipauto-core/src/attribute_utils.rs

```diff
@@ -27,257 +36,276 @@ pub fn update_openapi_macro_attributes(
+    for meta in nested_attributes {
+        if let Meta::List(list) = meta {
+            if list.path.is_ident("components") {
+                let nested = list
+                    .parse_args_with(Punctuated::<Meta, Token![,]>::parse_terminated)
+                    .expect("Expected a list of attributes inside components(...)!");
+                for meta in nested {
+                    if let Meta::List(list) = meta {
+                        if list.path.is_ident("schemas") {
+                            return list.tokens;
+                        }
+                    }
+                }
+            }
+        }
     }
+    TokenStream::new()
 }
 
-fn extract_responses(src_uto_macro: String) -> String {
-    if src_uto_macro.contains("responses(") {
-        let responses = src_uto_macro.split("responses(").collect::<Vec<&str>>()[1];
-        let responses = responses.split(')').collect::<Vec<&str>>()[0];
-        responses.to_string()
-    } else {
-        "".to_string()
+fn extract_responses(nested_attributes: &Punctuated<Meta, Token![,]>) -> TokenStream {
+    for meta in nested_attributes {
+        if let Meta::List(list) = meta {
+            if list.path.is_ident("components") {
+                let nested = list
+                    .parse_args_with(Punctuated::<Meta, Token![,]>::parse_terminated)
+                    .expect("Expected a list of attributes inside components(...)!");
+                for meta in nested {
+                    if let Meta::List(list) = meta {
+                        if list.path.is_ident("responses") {
+                            return list.tokens;
+                        }
+                    }
+                }
+            }
+        }
     }
+    TokenStream::new()
 }
 #[cfg(test)]
 mod test {
+    use proc_macro2::TokenStream;
     use quote::ToTokens;
+    use syn::punctuated::Punctuated;
 
     #[test]
-    fn test_remove_paths() {
+    fn test_extract_paths() {
         assert_eq!(
-            super::remove_paths("description(test),paths(p1),info(test)".to_string()),
-            "description(test),info(test)".to_string()
+            super::extract_paths(&syn::parse_quote!(paths(p1))).to_string(),
+            "p1".to_string()
         );
     }
 
-    #[test]
-    fn test_extract_paths() {
-        assert_eq!(super::extract_paths("paths(p1)".to_string()), "p1".to_string());
-    }
-
     #[test]
     fn test_extract_paths_empty() {
-        assert_eq!(super::extract_paths("".to_string()), "".to_string());
+        assert_eq!(super::extract_paths(&Punctuated::new()).to_string(), "".to_string());
     }
 
     #[test]
     fn test_build_new_openapi_attributes() {
         assert_eq!(
             super::build_new_openapi_attributes(
-                "".to_string(),
-                &"./src".to_string(),
-                &"".to_string(),
-                &"".to_string(),
+                Punctuated::new(),
+                &quote::quote!(crate::api::test),
```

Comment:
I couldn't get `./src` to behave reasonably here (if you quote it as `"./src"`, it will be interpolated as a string literal, including quotation marks). I'm pretty sure after having worked through this that the path originally given wouldn't be what is passed here anyways, since it's only used for discovery and the result of discovery should be the resolved crate / module / item paths? But you might want to double-check that I didn't bork something here.

Reply:
Looks weird but fine to me.
At least I couldn't find anything broken in the tests or when I tested it locally so its probably fine.